### PR TITLE
Tree: performance optimization

### LINKED ISSUE #14880: [Feature Request] Optimize performance when there are thousands of nodes and renderAfterExpand is false
### Existing Component
是

### Component Name
tree

### Description
When renderAfterExpand = false and the tree has thousands of nodes, click on any node to trigger the handleClick function of the TreeNode. Since the tree.store.currentNode has changed, the is-current class of thousands of nodes will trigger an update, causing performance problems.
I will submit a pull request to fix this issue.
image
image

### PR DESCRIPTION
**Pull request by ChenZhuoSteve**
Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

- Closes #14880 

image
image

### CODE REVIEW

**Comment by ChenZhuoSteve** [comment]
This is a previous version of the demo:
https://jsfiddle.net/chenzhuoSteve/b85ym7fc/5/
This is the demo after optimization:
https://jsfiddle.net/chenzhuoSteve/b85ym7fc/4

File: packages/tree/src/model/tree-store.js [hunk 1]
```diff
@@ -310,8 +310,11 @@ export default class TreeStore {
     return this.currentNode;
   }
 
-  setCurrentNode(node) {
-    this.currentNode = node;
+  setCurrentNode(currentNode) {
```
Comment (ziyoung) [inline packages/tree/src/model/tree-store.js]: 这样写更简洁一点。没有必要再对数据重新遍历一下。
```js
const prevCurrentNode = this.currentNode;
if (prevCurrentNode) {
  prevCurrentNode.isCurrent = false;
}
this.currentNode = currentNode;
this.currentNode.isCurrent = true;
```

**Comment by ziyoung** [comment]
@ChenZhuoSteve rebase 之后强推。

File: packages/tree/src/model/tree-store.js [hunk 2]
```diff
@@ -310,8 +310,13 @@ export default class TreeStore {
...
+    const prevCurrentNode = this.currentNode;
+    if (prevCurrentNode) {
+      prevCurrentNode.isCurrent = false;
+    }
+    this.currentNode = currentNode;
+    this.currentNode.isCurrent = true;
   }
 
   setUserCurrentNode(node) {
```
Comment (ziyoung) [inline packages/tree/src/model/tree-store.js]:
```js
setCurrentNodeKey(key) {
  if (key === null) {
    this.currentNode = null;
    return;
  }
  const node = this.getNode(key);
  if (node) {
    // this.currentNode = node;
    this.setCurrentNode(node);
  }
}
```

Comment (ziyoung) [reply packages/tree/src/model/tree-store.js]: @ChenZhuoSteve setCurrentNodeKey 的实现也调整一下。

Comment (ziyoung) [reply packages/tree/src/model/tree-store.js]: 注释是说明问题，提交的代码就不要包含注释了。😂

**Comment by ChenZhuoSteve** [comment]
@ziyoung 
thanks

**Pull request merged**